source/fe/fe_poly_tensor.cc

```diff
@@ -1180,12 +1180,12 @@ FE_PolyTensor<dim, spacedim>::fill_fe_face_values(
   // to take (all data sets for all
   // faces are stored contiguously)
 
-  const auto offset =
-    QProjector<dim>::DataSetDescriptor::face(this->reference_cell(),
-                                             face_no,
-                                             cell->combined_face_orientation(
-                                               face_no),
-                                             n_q_points);
+  const auto offset = QProjector<dim>::DataSetDescriptor::face(
+    this->reference_cell(),
+    face_no,
+    dim == 2 ? numbers::default_geometric_orientation :
+               cell->combined_face_orientation(face_no),
```

Comment:
One more thing: we support Hdiv and Hcurl 2d elements on quadrilateral meshes with non-standard orientations, so we don't have to make this assumption. See #18306 for more information.